beep/tests/test_maccor_to_biologic_mb.py

```diff
@@ -409,6 +415,83 @@ def test_discharge_current_step_conversion(self):
         self.single_step_to_single_seq_test(xml, diff_dict)
         pass
 
+    @unittest.skip
+    def test_header_insertion(self):
+        converter = MaccorToBiologicMb()
+
+        def set_i_range(tech_num, seq, idx):
+            seq_copy = copy.deepcopy(seq)
+            seq_copy["I Range"] = "100 mA"
+            return seq_copy
+        converter.seq_mappers.append(set_i_range)
+        converter.min_voltage_v = 0
+        converter.max_voltage_v = 4.45
+
+        converter.convert("/Users/patrickherring/Code/beep/beep/protocol/procedure_templates/formation_061621.000",
+                          "/Users/patrickherring/Code/beep/beep/protocol/biologic_templates", "formation_061621")
+
+        print(converter._mps_header_template[225:259])
+        self.assertEqual(converter._mps_header_template[225:260], '\tEcell min = {}\r\n\tEcell max = {}\r\n\t')
+
+    def test_conversion_with_updated(self):
```

Comment:
This test feels a little obtuse to me, it looks like we're doing a lot of stuff to essentially test whether the headers are being formatted correctly when the global voltage limits are set. Is that right or does this test something else?

If not consider just loading up any diagnostic we plan to hang onto, convert it, and check the header. No need to even set the I Range

Reply:
I will write this into the other test and delete this test

Reply:
Were you going to delete this?

Reply:
I deleted the other test that was just changing the header, this is the test that can run with the existing files

Reply:
I still don't quite understand why this test isn't
```python
converter = MaccorToBiologicMb()
converter.min_voltage_v = 2.0
converter.max_voltage_v = 4.4
def set_i_range(tech_num, seq, idx):
  seq_copy = copy.deepcopy(seq)
  seq_copy["I Range"] = "100 mA"
  return seq_copy

converter.convert(
  os.path.join(scratch_dir, "BioTest_000001.000"),
  TEST_FILE_DIR,
  "BioTest_000001"
 )
 
 self.assertEqual(converter._mps_header_template[225:260], '\tEcell min = {}\r\n\tEcell max = {}\r\n\t')
 ```
could you explain what all the procedure generation code is getting us?

Reply:
The generation code here is just to create a version of the protocol that can be tested with cells that are currently connected to the cycler. As such, I wanted to expose the parameters that we are using so that modifications can be made easily.

Reply:
Ok so if I understand correctly this test is to make sure that with the latest up-to-date generated maccor protocols the converter works.

I think that's fine but wonky enough that we should add a comment because it's very non-obvious looking at the test.

Reply:
Oh, the test that you have in the comment is actually already gone.

Reply:
`test_conversion_with_updated` is still appearing to me in the diff

Reply:
yes, that was the test I wanted to keep, but the code you have in the comment should not be there anymore

Reply:
I think there's some confusion over what code is confusing me, lines 423-461 of `test_maccor_to_biologic.py` in the latest commit of this branch `043d630388a997b24e3b3cf1e3e337f634d5e6f3`. I left a comment elsewhere highlighting the confusing code.

Reply:
Got it.